### DISCUSSION TITLE
Feedback and Suggestions

### DISCUSSION THREAD

**Discussion by pwt** [pwt/docker-airconnect-arm#10] [Q&A]
Feel free to use this issue to leave feedback and make suggestions.

**Comment by polyzen** [comment]
Hi @pwt, I'm wondering why `1000:1000:f` is used as the default latency here when `1000:2000` is suggested in AirConnect's readme for Sonos.

**Comment by pwt** [comment]
Hi @polyzen. It's just based on experience, really: I've used AirConnect extensively in multiple network environments and I find that latency settings of `750:750:f` provide very reliable buffering with Sonos. I make it `1000:1000:f` as the default for the Docker image just to be conservative.

You can adjust this back to the recommended defaults using `-e LATENCY='1000:2000'` (or whatever else you wish) on the Docker command line to start the container. See the 'Other configuration settings' section in the README.

The `:f` parameter just makes sure that silence frames are sent if the incoming stream is interrupted. This is to mitigate against the speaker closing its connection if there's a temporary interruption on the sending side.

**Comment by jlambert121** [comment]
Just wanted to leave some feedback - thanks for this!  Worked exactly as expected and I didn't have to write the Dockerfile or build the image!

**Comment by pwt** [comment]
> Just wanted to leave some feedback - thanks for this! Worked exactly as expected and I didn't have to write the Dockerfile or build the image!

Great! Glad it's useful. Thanks for the feedback.

**Comment by unlikeother** [comment]
Hello! I installed it on a rasperberry pi zero, and it seems to have been properly installed. However, none of my iOS devices detect my paired Google Home minis. What can i do?

**Reply by pwt** [reply]
Hi. I've never had any luck running this on a Pi Zero, whether as a native binary or containerised, hence it's not on my list of confirmed tested devices. I haven't explored extensively, however.

I should note that AirCast support is something of an afterthought, since I have no AirCast devices with which to test.

You might want to hop over to https://github.com/philippe44/AirConnect, and see if you can get one of the native `aircast-arm` binaries working (they're in the `bin` directory). If you get one working, let me know -- thanks.

**Comment by unlikeother** [comment]
Hello,

Thanks for the quick response! Thing is, once i got it working without
docker, my google mini was detected, and i could play from iOS (but sound
sometimes stopped). When i tried to upgrade my installation, i must have
done something wrong, and it stopped working. Also, it seems when i first
installed it, i only had to run airupnp-arm, and that makes no sense to me
now, because google mini is supposedly an aircast device. I eventually also
installed aircast-arm directly, and it also didn´t work.

Then i found your solution, and i´m trying it now. As far as i can tell,
the container is running normally, however no devices are detected...

pwt ***@***.***> escreveu no dia domingo, 3/10/2021 à(s)
18:07:

**Comment by unlikeother** [comment]
Forget it...i managed to solve my problem with this tutorial...i think i
had a permissions problem: https://jayden-chua.medium.com/free-airplay

antonioj ***@***.***> escreveu no dia domingo, 3/10/2021 à(s) 18:32:

**Comment by pwt** [comment]
Great! Thanks for following up.

For the benefit of others, the correct link to the article is: https://jayden-chua.medium.com/free-airplay-support-for-sonos-network-d91a9137593b

What was the specific permissions issue you had to fix?

**Comment by unlikeother** [comment]
This:

$ chmod +x /etc/airconnect/bin/airupnp-x86-64

That´s why it didn´t work, it wasn’t executable. However, i am having
problems making it work automatically…it seems to go down after a while,
and does not start when i reboot, despite having done what is stated on the
article to do so.

pwt ***@***.***> escreveu em seg., 4/10/2021 às 13:13 :

**Comment by pwt** [comment]
> This: $ chmod +x /etc/airconnect/bin/airupnp-x86-64 That´s why it didn´t work, it wasn’t executable. 

Thanks. I don't understand how you managed to run the x86 executable on a Pi Zero -- it shouldn't run at all, so I'm a little confused.

In the meantime, I've been modifying and testing my container image so it'll run on a Pi Zero. I should be able to release it in a day or two, subject to successful testing.

**Reply by pwt** [reply]
> In the meantime, I've been modifying and testing my container image so it'll run on a Pi Zero. I should be able to release it in a day or two, subject to successful testing.

@unlikeother : If you pull the updated image, it should now support the Pi Zero. The update instructions are in the README if you need them.

Let me know how it goes.

**Comment by unlikeother** [comment]
I've just copy pasted 😅 in my case it was aircast-arm ;)

Cool! I'll test it for sure

pwt ***@***.***> escreveu em seg., 4/10/2021 às 21:14 :

**Comment by unlikeother** [comment]
I have tested the Pi Zero supporting, and it´s working! However, how i can
adjust the buffering?

antonioj ***@***.***> escreveu em seg., 4/10/2021 às 21:54 :

**Reply by pwt** [reply]
The default buffering for the container is set to `1000:1000:f`, meaning 1s buffering on the input side, and 1s buffering on the output side. If you want to adjust this to `2000:2000:f` (for example), add the following to command line when you start the container:

`-e LATENCY='2000:2000:f'`

The better your network, the lower the latency required.

See the [Other configuration settings](https://github.com/pwt/docker-airconnect-arm#other-configuration-settings) section of the README for more details.

**Comment by unlikeother** [comment]
Thanks! Just what I needed :)

pwt ***@***.***> escreveu em qui., 7/10/2021 às 19:24 :
